packages/compiler-sfc/src/script/defineProps.ts

```diff
@@ -274,8 +274,16 @@ function genRuntimePropFromType(
       defaultString
     ])} }`
   } else {
-    // production: checks are useless
-    return `${finalKey}: ${defaultString ? `{ ${defaultString} }` : `{}`}`
+    if(defaultString){
+      return `${finalKey}: ${`{ ${defaultString} }`}`
+    }else{
+      if(/\.ce\.vue$/.test(ctx.filename)){
```

Comment:
Maybe we need to introduce an option `customElement` for identifying if it's a component of custom element.

Reply:
I get it 💪🏻